recipes/devices/cm4.sh

```diff
@@ -346,54 +403,79 @@ device_chroot_tweaks_pre() {
-		"imgpart=/dev/mmcblk0p2" "imgfile=/volumio_current.sqsh"
-		# Wait for root device
-		"rootwait" "bootdelay=5"
-		# I/O scheduler
-		"elevator=noop"
-		# Disable linux logo during boot
-		"logo.nologo"
-		# Disable cursor
-		"vt.global_cursor_default=0"
-		# Force predictable WLAN name
-		"net.ifnames=0"
-		# ALSA settings
-		"snd-bcm2835.enable_compat_alsa=1"
 	)
-
+	# Prepare Volumio splash, quiet, debug and loglevel.
+	# In init, "splash" controls Volumio logo, but in debug mode it will still be present
+	# In init, "quiet" had no influence (unused), but in init{v2,v3} it will prevent initrd console output
+	# So, when debugging, remove it and update loglevel to value: 8
+	if [[ $DEBUG_IMAGE == yes ]]; then
+		log "Debug image: remove splash from cmdline.txt" "cfg"
+		SHOW_SPLASH="nosplash" # Debug removed
+		log "Debug image: remove quiet from cmdline.txt" "cfg"
+		KERNEL_QUIET="" # Debug removed
+		log "Debug image: change loglevel to value: 8, debug, break and kmsg in cmdline.txt" "cfg"
+		KERNEL_LOGLEVEL="loglevel=8 debug break= use_kmsg=yes" # Default Debug
+	else
+		log "Default image: add splash to cmdline.txt" "cfg"
+		SHOW_SPLASH="splash" # Default splash enabled
+		log "Default image: add quiet to cmdline.txt" "cfg"
+		KERNEL_QUIET="quiet" # Default quiet enabled
+		log "Default image: change loglevel to value: 0, nodebug, no break  and no kmsg in cmdline.txt" "cfg"
+		KERNEL_LOGLEVEL="loglevel=0 nodebug use_kmsg=no" # Default to KERN_EMERG
+	fi
+	kernel_params+=("${SHOW_SPLASH}")
+	kernel_params+=("${KERNEL_QUIET}")
+
+	# Boot screen stuff
+	kernel_params+=("plymouth.ignore-serial-consoles")
+	# Raspi USB controller params
+	# TODO: Check if still required!
+	# Prevent Preempt-RT lock up
+	kernel_params+=("dwc_otg.fiq_enable=1" "dwc_otg.fiq_fsm_enable=1" "dwc_otg.fiq_fsm_mask=0xF" "dwc_otg.nak_holdoff=1")
+	# Output console device and options.
+	kernel_params+=("console=serial0,115200" "console=tty1")
+	# Image params
+	kernel_params+=("imgpart=UUID=${UUID_IMG} imgfile=/volumio_current.sqsh bootpart=UUID=${UUID_BOOT} datapart=UUID=${UUID_DATA} uuidconfig=cmdline.txt")
+	# A quirk of Linux on ARM that may result in suboptimal performance
+	kernel_params+=("pcie_aspm=off" "pci=pcie_bus_safe")
+	# Wait for root device
+	kernel_params+=("rootwait" "bootdelay=5")
+	# Disable linux logo during boot
+	kernel_params+=("logo.nologo")
+	# Disable cursor
+	kernel_params+=("vt.global_cursor_default=0")
+
+	# Buster tweaks
+	DISABLE_PN="net.ifnames=0"
+	kernel_params+=("${DISABLE_PN}")
+	# ALSA tweaks
+	kernel_params+=("snd-bcm2835.enable_compat_alsa=1")
+
+	# Further debug changes
 	if [[ $DEBUG_IMAGE == yes ]]; then
-		log "Creating debug image" "wrn"
-		log "Adding Serial Debug parameters"
+		log "Creating debug image" "dbg"
+		log "Adding Serial Debug parameters" "dbg"
 		echo "include debug.txt" >>/boot/config.txt
 		cat <<-EOF >/boot/debug.txt
 			# Enable serial console for boot debugging
 			enable_uart=1
+			dtoverlay=pi3-miniuart-bt
```

Comment:
this should not be in the CM4 recipe

Reply:
This should not be here: pi3-miniuart-bt